compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1082,8 +1082,18 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             && let Some(hir::Node::Item(hir::Item { span, ident, .. })) =
                                 hir.get_if_local(*def_id)
                         {
+                            let sm = self.tcx.sess.source_map();
+                            let span = sm
+                                .span_extend_to_prev_str(span.shrink_to_hi(), "fn", false, false)
```

Comment:
Please avoid using this family of function and only use them as a last resort. The source map is not really compatible with incremental compilation, and a possible source of bugs.

The issue is to put the `async` after the `pub`. The `pub`'s span is called `vis_span`. Can it be implemented using it?

Reply:
@cjgillot 
Thank you so much for your review!

I could implement the same functionality by using `vis_span`, and it's much clearer.
Sorry for having missed the `vis_span`.